### PR TITLE
[1.19] Add Ability for Recipes to define Item Remainders only for that Recipe

### PR DESCRIPTION
**Pull request by Dragon-Seeker**
Bonus is a Deep Copy Method for Item.Settings

### CODE REVIEW

File: src/main/java/io/wispforest/owo/mixin/recipe/RecipeManagerMixin.java
Comment (gliscowo) [inline src/main/java/io/wispforest/owo/mixin/recipe/RecipeManagerMixin.java]: It's spelled `oωo`

File: src/main/java/io/wispforest/owo/mixin/recipe/RecipeManagerMixin.java
Comment (gliscowo) [inline src/main/java/io/wispforest/owo/mixin/recipe/RecipeManagerMixin.java]: `JsonHelper.asItem()`

File: src/main/java/io/wispforest/owo/mixin/recipe/RecipeMixin.java
Comment (gliscowo) [inline src/main/java/io/wispforest/owo/mixin/recipe/RecipeMixin.java]: Please turn this into a return guard to reduce nesting

File: src/main/java/io/wispforest/owo/mixin/recipe/RecipeMixin.java
Comment (gliscowo) [inline src/main/java/io/wispforest/owo/mixin/recipe/RecipeMixin.java]: `var` time

File: src/main/java/io/wispforest/owo/util/RecipeSpecificRemainderStorage.java
Comment (gliscowo) [inline src/main/java/io/wispforest/owo/util/RecipeSpecificRemainderStorage.java]: The convention for the ID shorthand is `Id`, as such:  `recipeID` -> `recipeId`

File: src/main/java/io/wispforest/owo/mixin/recipe/RecipeManagerMixin.java
Comment (gliscowo) [inline src/main/java/io/wispforest/owo/mixin/recipe/RecipeManagerMixin.java]: This `if` should ideally be inverted and throw immediately if `item` is not present - avoiding one level of nesting

File: src/main/java/io/wispforest/owo/util/RecipeSpecificRemainderStorage.java
Comment (gliscowo) [inline src/main/java/io/wispforest/owo/util/RecipeSpecificRemainderStorage.java]: This class should be renamed to `RecipeSpecificRemainders`. Additonally, for consistency, all methods in it which specify further than `Remainer` should be shortened: `registerRecipeSpecificRemainder` -> `add`, `hasRecipeRemainders` -> `hasRemainders`

**Review by gliscowo — CHANGES_REQUESTED** [review]
Implementation looks fine, yet once again there are some general codestyle problems that need to be addressed before I will give approval

File: src/main/java/io/wispforest/owo/mixin/recipe/RecipeManagerMixin.java
Comment (Dragon-Seeker) [reply src/main/java/io/wispforest/owo/mixin/recipe/RecipeManagerMixin.java]: Not possible with how the API makes remainders: 
```
"owo:remainders": {
    "minecraft:sand": {
      "item": "minecraft:sand"
    }
  },
```

**Comment by Dragon-Seeker** [comment]
Would it be a good idea to have default behavior that if a empty JsonObject was given for the remainder item, it would default to use the given item as the Remainder?

File: src/main/java/io/wispforest/owo/mixin/recipe/RecipeManagerMixin.java
Comment (Dragon-Seeker) [reply src/main/java/io/wispforest/owo/mixin/recipe/RecipeManagerMixin.java]: Just slapped it into a JsonPrimitive and ran it into such

File: src/main/java/io/wispforest/owo/mixin/recipe/RecipeManagerMixin.java
Comment (Dragon-Seeker) [reply src/main/java/io/wispforest/owo/mixin/recipe/RecipeManagerMixin.java]: OwO

File: src/main/java/io/wispforest/owo/mixin/recipe/RecipeManagerMixin.java
Comment (Dragon-Seeker) [reply src/main/java/io/wispforest/owo/mixin/recipe/RecipeManagerMixin.java]: Fixed with using JsonHelper.getItem()

**Comment by Dragon-Seeker** [comment]
Could be such for specifying between the two: 

```
"owo:remainders": {
    "minecraft:sand": {
      "item": "minecraft:sand"
    },
   ...
   "item": "minecraft:sand"
  },
```

**Comment by Noaaan** [comment]
Might keep an eye on this PR to Fabric API, as it might supersede this: https://github.com/FabricMC/fabric/pull/2464

**Pull request closed without merging**
